torch/distributed/elastic/multiprocessing/tail_log.py

```diff
@@ -86,6 +86,7 @@ def __init__(
         name: str,
         log_files: Dict[int, str],
         dst: TextIO,
+        log_line_headers: Dict[int, str],
```

Comment:
1. Same here, for API BC, make `log_line_headers: Optional[Dict[int, str]] = None`.
2. Since all local procs should have the same prefix template, it might be cleaner to pass a map of `local_rank -> rank` and a string template `log_line_prefix` and do the substitution locally in `TailLog`, rather than making the API take a per-local-rank header.